### ISSUE TITLE
Switch Profitabilty Threshold

### ISSUE THREAD

**Issue by frederikvanduuren** [Sniffdog/Sniffdogminer#17]
hi,

instead of switching after x Secs, could it be possible to switch if profitibality is above a certain treshold?

keep up the good work ;)

fred

**Comment by dabrahost** [comment]
It would be a great feature! Algo switches even if profitability differs only 0,5-1%, it would be more profitable to stay on previous algo.

**Comment by pokepud3** [comment]
This would make an awesome feature. Less switching means it'll have a few seconds less startup time between algos and more mined overall. Realistically it could be a -profitability flag or something then you enter percentage. The miner is great as is but this would make it ever greater.

**Comment by frederikvanduuren** [comment]
i use awesome miner with this feature ;)

**Comment by Sniffdog** [comment]
Yes but we are not awesome miner lol

We are making notes for the new year ;)

**Comment by Slickmojo** [comment]
I love your work, thank you!  I would really like to have this idea implemented.

**Comment by Sniffdog** [comment]
Yes Slickmojo this is an interesting undertaking and has been noted.... I remember back in 2013 there was a program called Minercontrol that had this option... 

We are very busy in Jan/Feb with our company but believe me Sniffdog is the backbone!

**Comment by vatrak** [comment]
+1
Megaminer has this already where you can specify a % where it won't keep switching coins if the profits are close enough between them.
Another feature it has that would be cool on Sniffdogminer would be to let the last miner run for X seconds to give some time for the next miner to start up.
For some algos it takes 15-20 seconds for the miner to start working, so you can keep mining the old coin while it does that.

**Comment by Sniffdog** [comment]
I'm sure that if I looked hard enough online... I would find where megaminer found it... ;)

Second... -Delay (in seconds) allows the work to complete for old algo in startsniffin.bat

Lastly, did you see what I just forked? Right out in plain public view... someone placed it there for all to fork... the template for future things to come ;)

**Comment by frederikvanduuren** [comment]
where?

**Comment by frederikvanduuren** [comment]
what i am doing is running 2 seperate scripts...
one rescan set at 47 seconds & one set at 300, because 47 is a prime number, it never restarts at the same time ;)
so when they both run, they split the resources 50/50
when one restarts, the other gets the full GPU attention 100%

well thats the logic :)

**Comment by Sniffdog** [comment]
In my repository at github

**Comment by frederikvanduuren** [comment]
@Sniffdog you mean your zpool fork?
i must be blind ;) can you post the url? please
thanx

fred

**Comment by Sniffdog** [comment]
it is nicehashlegacy forked to make zpoollegacy 

https://github.com/Sniffdog/zPoolMiner

It's a hacked copy of nicehashminer.....lol 

I have documented all commits and timestamped my download... (nicehash has copyright laws for distribution) and documented bitcointalk thread where I found the link for the download and timestamped that too!

..but since it was changed by someone else then made public at github with no copyright attachments (at time of download) then I was able to fork...

everything is transparent on my end :)

**Comment by frederikvanduuren** [comment]
wow nice :) 

but why zpool?

**Comment by Sniffdog** [comment]
not sure but meh... since the release of Hashsynminer... this miner is kinda useless for us

**Comment by Sniffdog** [comment]
Ok!! Finally had time to download Multipoolminer and see what the heck you all are talking about and after a full day of benchmarking and throwing out crap from that miner...I finally got it working.... xD
   
If you have downloaded the newest version of sniffdog you will see how many coins are in each algo... this drop percentage probability estimates and throws off Multipoolminer probability threshold...

...seriously.. Multipoolminer had me running myr-gr... while my other rig was mining phi with Sniffdog...

I really think profitability would work really good with Miningpoolhub dues to better accuracy but running in a pool with (undecided) coins in a algo would kinda defeat the purpose...

prove me wrong?

**Comment by vatrak** [comment]
Nobody mentionned MPM... so i'm not sure what you mean.

Megaminer is the one where you can set a % so it doesn't keep switching algo when they are really close in profits.
And you can set a 15-20s period where it will start mining the next coin, but keep mining the current one and then close it after that period of time is done, so you don't end up wasting that 15-20sec waiting for the miner to actually start mining.

It's not a big deal for most algo that take 5-10seconds to start up, but for some of them that take 15+ seconds, it end up being a huge waste of time.

**Comment by Sniffdog** [comment]
Sooooo.... I spent all day yesterday checking out and fixing multipoolminer for nothing lol

Oh well :) 

It takes over 30 seconds for alexis to fire up on all pistons(cards)... Maybe why I may increase benching and my own personal interval a little higher

**Comment by Archdemon** [comment]
Yeah Profit % would be nice to set so you dont end up switching for little to nothing in terms of change.
Currently running klaust for neo (which isnt the fastest) and some Phi which also builds up really slow, benched at 65 but ended up at 102 after running for a couple of min.

No big deal since i changed it manually

**Comment by Sniffdog** [comment]
your right klaust is not the fastest but excavator is but I can't have it so that it's distributed

Everything has been noted... even for myself I would like to have Sniffdog able to (sync) together hashing of multiple machines no matter where they are in the world

**Comment by Archdemon** [comment]
Really? Running 150 core 450 mem on 1070s and always get best benches on klaust for neo.

-i 17.6
